schema/core.go

```diff
@@ -175,32 +175,52 @@ func (a CoreAttribute) validate(attribute interface{}) (interface{}, *errors.Sci
 		return nil, nil
 	}
 
-	if a.multiValued {
-		// return false if the multivalued attribute is not a slice.
-		arr, ok := attribute.([]interface{})
-		if !ok {
-			return nil, &errors.ScimErrorInvalidSyntax
+	if !a.multiValued {
+		return a.validateSingular(attribute)
+	}
+
+	switch arr := attribute.(type) {
+	case map[string]interface{}:
+		// return false if the multivalued attribute is empty.
+		if a.required && len(arr) == 0 {
+			return nil, &errors.ScimErrorInvalidValue
 		}
 
+		validMap := make(map[string]interface{}, len(arr))
+		for k, v := range arr {
+			for _, sub := range a.subAttributes {
+				if !strings.EqualFold(sub.name, k) {
+					continue
+				}
+				_, scimErr := sub.validate(v)
+				if scimErr != nil {
+					return nil, scimErr
+				}
+				validMap[sub.name] = v
+			}
+		}
+		return validMap, nil
+
+	case []interface{}:
 		// return false if the multivalued attribute is empty.
 		if a.required && len(arr) == 0 {
 			return nil, &errors.ScimErrorInvalidValue
 		}
 
-		attributes := make([]interface{}, 0)
-		for _, ele := range arr {
+		attributes := make([]interface{}, len(arr))
```

Comment:
Same note a above.

Reply:
I'd like to keep it that way. Valid `attributes` must be same length as `len(arr)` because `scimErr` will return soon if `ele` is not valid.

And it is more cost effective to determine the length of the array first.